Extension/src/LanguageServer/client.ts

```diff
@@ -2731,11 +2737,22 @@ export class DefaultClient implements Client {
             } else {
                 this.showConfigureIntelliSenseButton = false;
             }
-            ui.ShowConfigureIntelliSenseButton(this.showConfigureIntelliSenseButton, this);
         } else if (showConfigStatus && !displayedSelectCompiler) {
             this.promptSelectIntelliSenseConfiguration(false, "notification");
             displayedSelectCompiler = true;
         }
+
+        const configProviderType: ConfigurationType = this.configuration.ConfigProviderAutoSelected ? ConfigurationType.AutoConfigProvider : ConfigurationType.ConfigProvider;
```

Comment:
I think the telemetry should not be sent for the config provider if the config provider hasn't been registered yet and it's relying on the cache -- i.e. I think a 2nd configProviderNotSet variable should be saved that excludes the cache check and that should be checked for sending config provider telemetry. Otherwise, it sends invalid results early on.

Reply:
Like this?
```
const configProviderSet = settings.defaultConfigurationProvider ||
        this.configuration.CurrentConfiguration?.configurationProvider ||
        this.configuration.CurrentConfiguration?.configurationProviderInCppPropertiesJson;
```

Reply:
I was expecting the telemetry to not be sent if configProviderNotSet is true and configProviderNotSetAndNoCache is false -- otherwise it sends notConfigured.

Reply:
I can't think of any cases where we wouldn't want to send telemetry. Sending notConfigured in this case sounds correct (unless we auto-selected a compiler).

Reply:
The potential problem is that when users are auto configured, we'll always send notConfigured when in fact they'll use the cache and then it'll be auto configured after the config provider registers. Its seems better to exclude telemetry in this case since it's just noise.

Reply:
The queries we'll write for this won't care about that noise.  If the config provider never registers, then they are actually not configured.

Reply:
ok